### PR TITLE
Wip pathset

### PR DESCRIPTION
**Pull request by joshwalong**

### CODE REVIEW

File: binary-codec/main_test.go [hunk 1]
```diff
@@ -433,6 +433,63 @@ func TestEncode(t *testing.T) {
 			output:      "EA7C0F04C4D46544659A2D58525043686174E1",
 			expectedErr: nil,
 		},
+		{
+			description: "serialize Paths correctly from Tx3",
+			fromTx:      Tx3,
```
Comment (JCrawsh) [inline binary-codec/main_test.go]: Can we get rid of these now?

File: binary-codec/types/amount.go [hunk 2]
```diff
@@ -231,7 +231,7 @@ func SerializeIssuedCurrencyValue(value string) ([]byte, error) {
 }
 
 // Serializes an issued currency code to its bytes representation. The currency code can be 3 allowed string characters, or 20 bytes of hex
-func serializeIssuedCurrencyCode(currency string) ([]byte, error) {
+func SerializeIssuedCurrencyCode(currency string) ([]byte, error) {
```
Comment (JCrawsh) [inline binary-codec/types/amount.go]: Is there a reason this is now being exported?

File: binary-codec/types/pathset.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	typeCurrency = 0x10
+	typeIssuer   = 0x20
+
+	pathsetEndByte    = 0x00
+	pathSeparatorByte = 0xFF
+)
+
+type PathSet struct{}
+
+type ErrInvalidPathSet struct {
```
Comment (JCrawsh) [inline binary-codec/types/pathset.go]: Could this be a var type error?

File: binary-codec/main_test.go (hunk 1 above)
Comment (joshwalong) [reply binary-codec/main_test.go]: removed the repeat cases

File: binary-codec/types/pathset.go (hunk 3 above)
Comment (joshwalong) [reply binary-codec/types/pathset.go]: yup changed ty

File: binary-codec/types/amount.go (hunk 2 above)
Comment (joshwalong) [reply binary-codec/types/amount.go]: not anymore! changed

File: binary-codec/main_test.go [hunk 4]
```diff
@@ -3,25 +3,20 @@ package binarycodec
...
-	LedgerEntryTypeExample = "110072"             // generated from the xrpl js project
-	UInt64TypeExample      = "34000000044B82FA09" // generated from the xrpl js project
-	UInt8IntExample        = "011019"             // generated from the xrpl js project
+	Tx1 = "120007220008000024001ABED82A2380BF2C2019001ABED764D55920AC9391400000000000000000000000000055534400000000000A20B3C85F482532A9578DBB3950B85CA06594D165400000037E11D60068400000000000000A732103EE83BB432547885C219634A1BC407A9DB0474145D69737D09CCDC63E1DEE7FE3744630440220143759437C04F7B61F012563AFE90D8DAFC46E86035E1D965A9CED282C97D4CE02204CFD241E86F17E011298FC1A39B63386C74306A5DE047E213B0F29EFA4571C2C8114DD76483FACDEE26E60D8A586BB58D09F27045C46"
```
Comment (JCrawsh) [inline binary-codec/main_test.go]: Could we also get rid of these?

**Review by JCrawsh — APPROVED** [review]
LGTM

File: binary-codec/main_test.go (hunk 4 above)
Comment (joshwalong) [reply binary-codec/main_test.go]: yep will do thanks

**Pull request merged**